src/protocol/attribution/accumulate_credit.rs

```diff
@@ -306,17 +297,17 @@ mod tests {
 
         let result = try_join!(h0_future, h1_future, h2_future).unwrap();
 
-        assert_eq!(result.0.len(), raw_input.len());
-        assert_eq!(result.1.len(), raw_input.len());
-        assert_eq!(result.2.len(), raw_input.len());
-
-        (0..(result.0.len())).for_each(|i| {
-            let v = validate_and_reconstruct((
-                result.0[i].credit,
-                result.1[i].credit,
-                result.2[i].credit,
-            ));
-            assert_eq!(v.as_u128(), expected_credit_output[i]);
-        });
+        assert_eq!(result.0.len(), RAW_INPUT.len());
+        assert_eq!(result.1.len(), RAW_INPUT.len());
+        assert_eq!(result.2.len(), RAW_INPUT.len());
+
+        for (i, expected) in EXPECTED.iter().enumerate() {
+            let v = validate_and_reconstruct(
```

Comment:
Major change to this function: it now takes three arguments *by reference*.  This is most ergonomic in the end.